neqo-transport/src/connection/mod.rs

```diff
@@ -987,6 +987,12 @@ impl Connection {
         }
     }
 
+    /// Whether the given [`ConnectionIdRef`] is a valid local [`ConnectionId`].
+    #[must_use]
+    pub fn is_valid_local_cid(&self, cid: ConnectionIdRef) -> bool {
+        self.cid_manager.is_valid(cid)
+    }
```

Comment:
Instead of the indirection, sniffing new `ConnectionId`s through the provided `ConnectionIdGenerator` wrapper, the `Server` now explicitly asks the `Connection` whether a `ConnectionId` is a valid id of the connection.